From 292dd07ebd79a5236921231d82c85bcb3874f188 Mon Sep 17 00:00:00 2001
From: Jelle van der Waa <jelle@vdwaa.nl>
Date: Thu, 19 Nov 2020 14:51:31 +0100
Subject: [PATCH] Handle Request.Host containing the server port

As written in the net/http/#Request documentation Host may be of the
form "host:port" and as defined in rfc4366 section-3.1 the serverName
should not contain the port or colon.
---
 requester/requester.go | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/requester/requester.go b/requester/requester.go
index fd7277e..924d945 100644
--- a/requester/requester.go
+++ b/requester/requester.go
@@ -20,6 +20,7 @@ import (
 	"crypto/tls"
 	"io"
 	"io/ioutil"
+	"net"
 	"net/http"
 	"net/http/httptrace"
 	"net/url"
@@ -235,10 +236,15 @@ func (b *Work) runWorkers() {
 	var wg sync.WaitGroup
 	wg.Add(b.C)
 
+	hostName, _, err := net.SplitHostPort(b.Request.Host);
+	if err != nil {
+		hostName = b.Request.Host;
+	}
+
 	tr := &http.Transport{
 		TLSClientConfig: &tls.Config{
 			InsecureSkipVerify: true,
-			ServerName:         b.Request.Host,
+			ServerName:         hostName,
 		},
 		MaxIdleConnsPerHost: min(b.C, maxIdleConn),
 		DisableCompression:  b.DisableCompression,
-- 
2.29.2

